kimchi/src/verifier_index.rs

```diff
@@ -238,7 +238,7 @@ where
             endo: self.cs.endo,
             lookup_index,
             linearization: self.linearization.clone(),
-            fr_sponge_params: self.cs.fr_sponge_params.clone(),
+            fr_sponge_params: self.srs.scalar_sponge_params.clone(),
```

Comment:
if this is present in the SRS, then we can remove it from the verifier index unless I'm missing something

Reply:
The problem with that is VerifierIndex::srs() that returns the srs or tries to create it, in that case VerifierIndex won't be able to provide the params to SRS::create(). VerifierIndex::srs() could take the params but I think that will just be pushing the problem.
But my proposal about using a trait should solve this too.